front-end/.storybook/preview.js

```diff
@@ -6,4 +8,12 @@ export const parameters = {
       date: /Date$/,
     },
   },
-}
\ No newline at end of file
+};
+
+export const decorators = [
+  (Story) => (
+    <MemoryRouter initialEntries={["/"]}>
```

Comment:
Storybook is a different environment than our main app where individual components can be "isolated" for a focused development.
Nevertheless, no component is truly ever independent. Components used in the main application have access to `react-router` global variables, so we need to mock the same set of global variables here in Storybook for the components to function.

Reply:
Note that in the main app, we use `BrowserRouter`, but here we use `MemoryRouter`. They have the same set of variables that the component use, just different under-the-hood implementation. BrowserRouter access the browser history API and enable real navigation with the back and forward buttons. MemoryRouter simply use an array, and that's all. Useful for mocking though, like this situation.